sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/LogicalPlanSuite.scala

```diff
@@ -145,4 +146,16 @@ class LogicalPlanSuite extends SparkFunSuite {
     assert(query.where(Literal.FalseLiteral).maxRows.contains(0))
     assert(query.where(Literal.FalseLiteral).maxRowsPerPartition.contains(0))
   }
+
+  test("SPARK-46285: foreachWithSubqueries") {
+    val input = UnresolvedRelation(Seq("subquery_table"))
+    val input2 = UnresolvedRelation(Seq("t"))
+    val plan = Project(Seq(Alias(ScalarSubquery(input), "s")()), input2)
```

Comment:
Super nit: this plan is actually invalid as `ScalarSubquery` requires the plan to return at least one row. Maybe `Exists` is better for testing, we can use it in `Filter`

Reply:
Dot it. Done.